### PR TITLE
[Azure sym server] Upload the DLLs

### PR DESCRIPTION
**Pull request by sebmarchand**
Upload the PE files generated by the ArcCore build to the Azure symbol server in addition to the PDBs.

Also make the symbol search pattern explicit as the default wasn't gathering all desired files.

### CODE REVIEW

File: .github/workflows/swift-toolchain.yml [hunk 1]
```diff
@@ -793,12 +793,22 @@ jobs:
           path: |
             ${{ github.workspace }}/BinaryCache/1/**/*.pdb
 
-      - uses: microsoft/action-publish-symbols@v2.1.6
+      - name: Upload PDBs to Azure
```
Comment (tristanlabelle) [inline .github/workflows/swift-toolchain.yml:796]: This is in the compilers job. I believe we need this in the sdk job (or both)

Comment (compnerd) [reply .github/workflows/swift-toolchain.yml:796]: Yes, we will want it in both. I'd really like to try to get the glob working though.

File: .github/workflows/swift-toolchain.yml [hunk 2]
```diff
@@ -793,12 +793,32 @@ jobs:
...
+        uses: microsoft/action-publish-symbols@v2.1.6
         if: ${{ needs.context.outputs.debug_info }}
         with:
           accountName: ${{ vars.SYMBOL_SERVER_ACCOUNT }}
           personalAccessToken: ${{ secrets.SYMBOL_SERVER_PAT }}
           symbolsFolder: ${{ github.workspace }}/BinaryCache/1
+          searchPattern: '**/bin/**/*.pdb'
```
Comment (tristanlabelle) [inline .github/workflows/swift-toolchain.yml:803]: @compnerd I made these explicit (from the action's default). Is this pattern correct, and should I follow the same format for dlls and exes? I don't have a toolchain build handy to check.

File: .github/workflows/swift-toolchain.yml (hunk 1 above)
Comment (tristanlabelle) [reply .github/workflows/swift-toolchain.yml:796]: Pushed a fix.

File: .github/workflows/swift-toolchain.yml (hunk 2 above)
Comment (tristanlabelle) [reply .github/workflows/swift-toolchain.yml:803]: Turns out `dispatch.dll`/`dispatch.pdb` are not emitted within a "bin" directory, so we are not uploading them. (fyi @hjyamauchi )

**Comment by tristanlabelle** [comment]
Trial run here: https://github.com/thebrowsercompany/swift-build/actions/runs/8332881460

**Review by compnerd — COMMENTED** [review]
I'm fine with this conceptually, but I think that I want to see if we can collapse these into a single run. That would reduce the overall impact of the step as we can upload more items in parallel/

File: .github/workflows/swift-toolchain.yml (hunk 2 above)
Comment (compnerd) [reply .github/workflows/swift-toolchain.yml]: Yeah, libdispatch not using `bin` for the output is why the default pattern didn't work.

**Comment by compnerd** [comment]
Doing a collapsed variant run at https://github.com/thebrowsercompany/swift-build/actions/runs/8333794115

**Review by compnerd — APPROVED** [review]
Well, this is unfortunate, both forms fail. We can take a look at this later I suppose, merging this to avoid blocking.

**Comment by tristanlabelle** [comment]
@compnerd We can't use a single invocation because https://github.com/microsoft/action-publish-symbols/issues/32

**Pull request merged**